lib/fontv/libfv.py

```diff
@@ -0,0 +1,299 @@
+        self.develop_string = develop
+        self.release_string = release
+        self.sha1_develop = sha1_develop
+        self.sha1_release = sha1_release
+
+        # fonttools TTFont for font on self.fontpath
+        # raises IOError if filepath does not exist
+        # raises fontTools.ttLib.TTLibError if fontpath is not a ttf or otf font
+        self.ttf = ttLib.TTFont(self.fontpath)
+
+        # data containers
+        self._nameID_5_dict = {}
+        self.version_string_parts = []
+        self.version = ""
+        self.status = ""
+        self.metadata = []
+
+        # truth test values for string contents
+        self.contains_metadata = False
+        self.contains_status = False
+        self.is_development = False
+        self.is_release = False
+
+        # object instantiation methods
+        self._read_version_string()  # read version string on fontpath at object instantiation
+
+    def _read_version_string(self):
+        """
+        _read_version_string is a private method that reads OpenType name table ID 5 data from otf and ttf fonts
+        and sets FontVersion object properties.  The method is called on instantiation of a FontVersion
+        object
+        :return: None
+        """
+
+        namerecord_list = self.ttf['name'].names
+        # read in name records
+        for record in namerecord_list:
+            if record.nameID == 5:
+                # map dictionary as {(platformID, platEncID, langID) : version string}
+                recordkey = (record.platformID, record.platEncID, record.langID)
+                self._nameID_5_dict[recordkey] = record.toUnicode()
+
+        # begin parsing of the version string
+        if (3, 1, 1033) in self._nameID_5_dict:
+            version_string = self._nameID_5_dict[(3, 1, 1033)]
+        elif (1, 0, 0) in self._nameID_5_dict:
+            version_string = self._nameID_5_dict[(1, 0, 0)]
+        else:
+            version_string = ""  # TODO: raise exception?
+
+        # parse version string into substrings
+        self._parse_version_substrings(version_string)
+
+        # define version as first substring
+        self.version = self.version_string_parts[0]
+
+        # metadata parsing
+        self._parse_metadata()         # parse the metadata
+        self._parse_status()           # parse the version substring dev/rel status indicator
+
+    def _get_repo_commit(self):
+        repo = Repo(get_git_root_path(self.fontpath))
+        gitpy = repo.git
+        # git rev-list --abbrev-commit --max-count=1 --format="%h" HEAD - abbreviated unique sha1 for the repository
+        # number of sha1 hex characters determined by git (addresses https://github.com/source-foundry/font-v/issues/2)
+        full_git_sha_string = gitpy.rev_list('--abbrev-commit', '--max-count=1', '--format="%h"', 'HEAD')
+        unicode_full_sha_string = tounicode(full_git_sha_string)
+        sha_string_list = unicode_full_sha_string.split("\n")
```

Comment:
Location of the fix for Windows git sha1 hash parsing issue.  This will need to be fixed in the executable module as well until we transition to use of the library in the font-v executable.